query-request.hh

```diff
@@ -372,4 +374,29 @@ public:
     friend std::ostream& operator<<(std::ostream& out, const read_command& r);
 };
 
+struct forward_request {
+    enum class reduction_type {
+        count,
+    };
+
+    // multiple reduction types are needed to support queries like:
+    // `SELECT min(x), max(x), avg(x) FROM tab`
+    std::vector<reduction_type> reduction_types;
+
+    query::read_command cmd;
+    dht::partition_range_vector pr;
+
+    db::consistency_level cl;
+    lowres_clock::time_point timeout;
+
+    friend std::ostream& operator<<(std::ostream& out, const forward_request& r);
+    friend std::ostream& operator<<(std::ostream& out, const reduction_type& r);
+};
```

Comment:
Something we are missing is WHERE clause forwarding. Do we check that the WHERE clause is empty?

We'll have to work on serializing expressions post-merge, so we can pass more complicated queries (both selectors and filters).